docs/deploy/vitis_ai.rst

```diff
@@ -0,0 +1,671 @@
+   # load the module into memory
+   loaded_json = open(temp.relpath("deploy_graph.json")).read()
+   loaded_lib = tvm.runtime.load_module(path_lib)
+   loaded_params = bytearray(open(temp.relpath("deploy_param.params"), "rb").read())
+
+   module = tvm.contrib.graph_runtime.create(loaded_json, loaded_lib, ctx)
+   module.load_params(loaded_params)
+   module.set_input(name, data)
+   module.run()
+
+Edge usage
+~~~~~~~~~~
+
+This section shows how to accelerate a convolutional neural network
+model in TVM with Vitis-AI at the edge. The first couple of steps will
+have to be run on the host machine and take care of quantization and
+compilation for deployment at the edge.
+
+Host steps
+^^^^^^^^^^
+
+To be able to target the Vitis-AI cloud DPUCZDX8G target we first have
+to import the target in PyXIR. This PyXIR package is the interface being
+used by TVM to integrate with the Vitis-AI stack. Additionaly, import
+the typical TVM and Relay modules and the Vitis-AI contrib module inside
+TVM.
+
+.. code:: bash
+
+   import pyxir
+   import pyxir.contrib.target.DPUCZDX8G
+
+   import tvm
+   import tvm.relay as relay
+   from tvm.contrib.target import vitis_ai
+   from tvm.relay.build_module import bind_params_by_name
+   from tvm.relay.op.contrib.vitis_ai import annotation
+
+After importing a convolutional neural network model using the usual
+Relay API's, annotate the Relay expression for the given Vitis-AI DPU
+target and partition the graph.
+
+.. code:: bash
+
+   mod["main"] = bind_params_by_name(mod["main"], params)
+   mod = annotation(mod, params, target)
+   mod = relay.transform.MergeCompilerRegions()(mod)
+   mod = relay.transform.PartitionGraph()(mod)
+
+Now, we can build the TVM runtime library for executing the model. The
+TVM target is 'llvm' as the operations that can't be handled by the DPU
+are executed on the CPU. At this point that means the CPU on the host.
+The Vitis-AI target is DPUCZDX8G-zcu104 as we are targeting the edge DPU
+on the ZCU104 board and this target is passed as a config to the TVM
+build call. Note that different identifiers can be passed for different
+targets, see `edge targets info <#edge-requirements>`__.
+
+.. code:: bash
+
+   tvm_target = 'llvm'
+   target='DPUCZDX8G-zcu104'
+
+   with tvm.transform.PassContext(opt_level=3, config= {'relay.ext.vitis_ai.options.target': target}):   
+      graph, lib, params = relay.build(mod, tvm_target, params=params)
+
+Additionaly, already build the deployment module for the ARM CPU target
+and serialize:
+
+.. code:: bash
+
+   # Export lib for aarch64 target
+
+   tvm_target = tvm.target.arm_cpu('ultra96')
+   lib_kwargs = {
+        'fcompile': contrib.cc.create_shared,
+        'cc': "/usr/aarch64-linux-gnu/bin/ld"
+   }
+
+   with tvm.transform.PassContext(opt_level=3,
+                                   config={'relay.ext.vitis_ai.options.target': target,
+                                           'relay.ext.vitis_ai.options.build_dir': target + '_build'}):
```

Comment:
do we need to have config for the build dir? can we just use a temporary dir to do this?